pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -353,6 +355,7 @@ impl RemoteTimelineClient {
     pub fn last_uploaded_consistent_lsn(&self) -> Option<Lsn> {
         match &*self.upload_queue.lock().unwrap() {
             UploadQueue::Uninitialized => None,
+            UploadQueue::Broken => None, // could we return something?
             UploadQueue::Initialized(q) => Some(q.last_uploaded_consistent_lsn),
             UploadQueue::Stopped(q) => Some(q.last_uploaded_consistent_lsn),
         }
```

Comment:
If we return `None` here, the walreceiver status reported to SKs will have `remote_consistent_lsn = 0` (follow call graph of this function upwards).

Before this PR, it certainly was never possible for SKs to see Lsn(0), because we only launch walreceiver once we have initialized the upload queue.

With this PR, I _think_ it's possible, because not all callers of stop guarantee that walreceiver is shut down.
Maybe I'm wrong, but, it would be exceptionally brittle.

So, we have to make sure SKs are OK with us returning None here.

The alternative, I guess, is to copy the last_uploaded_consistent_lsn into the `Broken` state.
I don't think that would be wrong (?)
Need to think about upload ups running concurrently with `stop()`...

WDYT? IMO `abort()` is preferable than messing up `last_uploaded_consistent_lsn`

Reply:
I am not yet on the map, currently wondering why this is not using RAII but all this mutable state. But poisoning the mutex helps here as well, so haven't yet found an issue.

Reply:
Yeah we could panic here, because a panic should had already poisoned the upload_queue.